### PR TITLE
ci: fix optimize deployment

### PR DESCRIPTION
**Pull request by misiekhardcore**
## Description

This PR fixes Optimzie CI deployment step

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: .github/workflows/optimize-ci.yml [hunk 1]
```diff
@@ -95,7 +95,7 @@ jobs:
         build-args: |
           VERSION=${{ steps.pom-info.outputs.project_version }}
           REVISION=${{ github.event.pull_request.head.sha }}
-        tags: "registry.camunda.cloud/team-optimize/optimize:pr-${{ github.event.pull_request.head.sha }}"
```
Comment (PHWaechtler) [inline .github/workflows/optimize-ci.yml:98]: Hm, looking at this again now I'm wondering if `team-optimize` is for internal use and `optimize-ee` just for actualy releases 🤔 Looks like the uses are similar in other teams as well. Can we double check this, maybe with other monorepo folk or infra?

Comment (misiekhardcore) [reply .github/workflows/optimize-ci.yml:98]: I think you are right, looking at other teams workflows they are using `team-xyz`

Comment (misiekhardcore) [reply .github/workflows/optimize-ci.yml:98]: actually no other team is using `xyz-ee`, but this I believe is because they are all in single jar already

Comment (PHWaechtler) [reply .github/workflows/optimize-ci.yml]: I can see the `tram-z` one being used in those `operate/tasklist-ci-build-reusable.yml`

**Review by HeleneW-dot — APPROVED** [review]

**Pull request merged**